src/controllers/google-controller.ts

```diff
@@ -0,0 +1,48 @@
+import { Request, Response, NextFunction } from 'express'
+import jwt from 'jsonwebtoken'
+
+import { User } from 'models'
+import { googleAuthSchema } from 'schemas'
+import { ErrorType } from 'types'
+
+export const googelAuth = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    await googleAuthSchema.validateAsync(req.body)
+
+    const username = req.body.email.replace('@gmail.com', '')
```

Comment:
i tried to use next auth google authentication and when i logged in user object returned name... so u have access to it, why not use that one?? instead of doing this

Reply:
In the assignment documentation, it said that the username had to be lowercase which is why I decided to do this. Should I change it?

Reply:
:thinking: okay here is the problem here, they were will be dots with this approach but with approach I said there will probably be spaces, in there... :thinking: :thinking: what do we doo, okay so do it with name, IMO it will look ugly as hell with dots in username, and space is fine, shouldn't cause any problems considering google users won't use username to login

Reply:
Okay, and should I make the names lower case or keep them the way they are?

Reply:
do as u wish